build.gradle

```diff
@@ -90,7 +93,28 @@ githubPages {
 }
 
 shadowJar {
-    classifier = 'all'
+    baseName = 'launchdarkly-client'
+    //no classifier means that the fat + shaded jar becomes the default artifact
+    classifier = ''
+
+    // Shade all jars except for launchdarkly
+    relocate('com', 'com.launchdarkly.shaded.com') {
```

Comment:
Hmm. I hadn't considered that when I made the suggestion to shade Gson. I think we should not shade it then-- leaking a shaded class into an external interface isn't a good practice. 

In addition to just having the wrong type, *interacting* with the JsonElement is going to be hard-- you'll have to manipulate the shaded API, which exposes more shaded footprint.